simexpal/base.py

```diff
@@ -522,6 +533,55 @@ def make_variation(prod):
 
 		return [make_variation(prod) for prod in itertools.product(*variation_bundle)]
 
+	def update_status_cache_dict(self, run):
+
+		def _get_status_entry(run):
+			try:
+				with open(run.output_file_path('status'), 'r') as f:
+					status_dict = yaml.load(f, Loader=YmlLoader)
+					last_mod = os.fstat(f.fileno()).st_mtime
+
+				if status_dict['timeout']:
+					return Status.TIMEOUT, last_mod
+				elif status_dict['signal']:
+					return Status.KILLED, last_mod
+				elif status_dict['status'] > 0:
+					return Status.FAILED, last_mod
+				return Status.FINISHED, last_mod
+			except FileNotFoundError:
+				pass
+			if os.access(run.output_file_path('out'), os.F_OK):
+				return Status.STARTED, os.stat(run.output_file_path('out')).st_mtime
+			elif os.access(run.aux_file_path('run'), os.F_OK):
+				return Status.SUBMITTED, os.stat(run.aux_file_path('run')).st_mtime
+			elif os.access(run.aux_file_path('lock'), os.F_OK):
+				return Status.IN_SUBMISSION, os.stat(run.aux_file_path('lock')).st_mtime
+			return Status.NOT_SUBMITTED, None
+
+		status, last_mod = _get_status_entry(run)
+
+		if run.experiment.display_name not in self.status_cache_dict:
+			# Stringify key tuple as json can not serialize pure tuples.
+			self.status_cache_dict[run.experiment.display_name] = {
```

Comment:
We use `display_name` here. That is probably not the best choice; we could add another `internal_name` (or similar) function that also incorporates the repetition. That way, we would not need a nested dict.

Reply:
Note that we also form `display_name` multiple times, which might involve more string concatenations / memory allocations than necessary (since I doubt that the Python interpreter is able to optimize these calls).

Reply:
Ok, I will add an ``Experiment.internal_name`` property that I will use for the cache entry keys.

Reply:
I added an ``internal_name`` property to the ``Run`` class. It works like ``Experiment.display_name``, but adds a suffix ``[<repetition>]``.